### PR TITLE
Update out-of-date dependencies, upgrade mongodb

### PR DESCRIPTION
**Pull request by amcgee**

### CODE REVIEW

File: lib/engines/mongodb.js [hunk 1]
```diff
@@ -67,31 +67,36 @@ MongoCollection.prototype.findOne = function(query, projection, callback) {
...
+	console.log( JSON.stringify( query ) );
 	return new MongoCursor( this.db.find(query, projection) );
 };
 MongoCollection.prototype.insert = function(obj, opt, callback) {
-	return this.db.insertAsync(obj, opt)
-		.then(function(arr) {
-			_.forEach(arr, function(item) {
+	return this.db.insertOneAsync(obj, opt)
+		.then(function(res) {
+			if ( res.result.ok != 1 )
```
Comment (ottumm) [inline lib/engines/mongodb.js]: What's our "if/then" style? Feels like open brace should be on the same line as the condition. /nitpick

**Comment by ottumm** [comment]
:+1:

**Comment by amcgee** [comment]
This introduces an unfortunate breaking change which requires permission filter objects to specify 'null' rather than 'undefined'.  The new mongodb engine ignores keys with 'undefined' values, wheras the previous engine would take this to mean that that key should be undefined within the mongodb object as well.  Thankfully unit tests caught this - see permissions.test.js::33.  I'm not sure if there's a good way to modify the query object within REM before passing it on to mongo, but if there is that would be preferred...

File: lib/engines/mongodb.js (hunk 1 above)
Comment (amcgee) [inline lib/engines/mongodb.js]: Yeah, I also sometimes omit the braces for one-line if bodies in javascript but I'd be happy moving away from that.

**Comment by ottumm** [comment]
Seems like we could enumerate through all the keys and filter them through ```foo.key || null``` before passing to MongoDB.

**Pull request merged**